TriangleNet_Engine/Compute/VoronoiRegions.cs

```diff
@@ -0,0 +1,313 @@
+            }
+
+            //Try fit plane if no is provided
+            if (plane == null)
+            {
+                plane = points.FitPlane(tolerance);
+            }
+
+            if (plane == null)
+            {
+                Engine.Reflection.Compute.RecordError("Could not fit a plane through the Points.");
+                return new List<Polyline>();
+            }
+
+            //Check all points within distance of the plane
+            if (points.Any(x => x.Distance(plane) > tolerance))
+            {
+                BH.Engine.Reflection.Compute.RecordError("Can only handle coplanar points, in the plane!");
+                return new List<Polyline>();
+            }
+
+            //Calculate the local coordinates
+            Vector localX = points[1] - points[0];
+            Vector localY = plane.Normal.CrossProduct(localX);
+            Point min = points.Min();
+            Point refPt = new Point { X = min.X, Y = min.Y };
+            Cartesian localSystem = Create.CartesianCoordinateSystem(min, localX, localY);
+            Cartesian globalSystem = Create.CartesianCoordinateSystem(refPt, Vector.XAxis, Vector.YAxis);
+
+            //Transform to xy-plane
+            List<Point> xyPoints = points.Select(x => x.Orient(localSystem, globalSystem)).ToList();
+
+
+            //Add point at all corners. This is to try to handle weirdness at the boundaries
+            BoundingBox bounds = xyPoints.Bounds();
+
+            double lengthX, lengthY;
+
+            if (boundarySize <= 0)
+            {
+                lengthX = (bounds.Max.X - bounds.Min.X);
+                lengthY = (bounds.Max.X - bounds.Min.X);
+            }
+            else
+            {
+                lengthX = boundarySize;
+                lengthY = boundarySize;
+            }
+            double minX = bounds.Min.X - lengthX;
+            double minY = bounds.Min.Y - lengthY;
+            double maxX = bounds.Max.X + lengthX;
+            double maxY = bounds.Max.Y + lengthY;
+
+            xyPoints.Add(new Point { X = minX, Y = minY });
+            xyPoints.Add(new Point { X = minX, Y = maxY });
+            xyPoints.Add(new Point { X = maxX, Y = maxY });
+            xyPoints.Add(new Point { X = maxX, Y = minY });
+
+            //Convert to TringleNet vertecies
+            List<TriangleNet.Geometry.Vertex> vertecies = xyPoints.Select(p => new TriangleNet.Geometry.Vertex(p.X, p.Y)).ToList();
+
+            // Triangulate
+            TriangleNet.Meshing.ITriangulator triangulator = new TriangleNet.Meshing.Algorithm.Dwyer();
+            TriangleNet.Configuration config = new TriangleNet.Configuration();
+            TriangleNet.Mesh mesh = (TriangleNet.Mesh)triangulator.Triangulate(vertecies, config);
+
+            TriangleNet.Voronoi.StandardVoronoi voronoi = new TriangleNet.Voronoi.StandardVoronoi(mesh);
+
+            // Convert regions to BHoMGeometry. Skip the last 4 faces as they correspond to the added boundary points.
```

Comment:
This comment looks like it actually relates to the `for` loop - the bit about skipping 4 at least, I would advise moving this comment down to be closer to the statement it refers to